frame/nomination-pools/src/migration.rs

```diff
@@ -724,4 +724,106 @@ pub mod v5 {
 			Ok(())
 		}
 	}
+
+	/// This migration summarizes and initializes the TotalValueLocked StorageValue for all Pools.
+	pub struct MigrateToV6<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> OnRuntimeUpgrade for MigrateToV6<T> {
+		fn on_runtime_upgrade() -> Weight {
+			let current = Pallet::<T>::current_storage_version();
+			let onchain = Pallet::<T>::on_chain_storage_version();
+
+			log!(
+				info,
+				"Running migration with current storage version {:?} / onchain {:?}",
+				current,
+				onchain
+			);
+
+			if current == 6 && onchain == 5 {
+				let mut migrated = 0u64;
+				let mut tvl: BalanceOf<T> = Zero::zero();
+				for (pool_id, _pool) in BondedPools::<T>::iter() {
+					migrated.saturating_inc();
+					if let Some(bonded_pool) = BondedPool::<T>::get(pool_id) {
+						tvl = tvl.saturating_add(
+							T::Staking::total_stake(&bonded_pool.bonded_account())
+								.unwrap_or_default(),
+						);
+					}
+				}
+				TotalValueLocked::<T>::set(tvl);
```

Comment:
When you need to derive or transform data in Rust please avoid imperative programming like this and make use of iterators instead. They are far easier to reason about and less bug prone.

If you're not sure where to start, chatgpt can be a helpful tool for learning about iterators and provide helpful hints for refactoring code.

Let me know if you have any questions about this.

Reply:
Thanks for the reminder. I rewrote it with more iterator usage and also optimized the storage access to be one less per migrated pool. 

`let migrated = BondedPools::<T>::count();
				let tvl: BalanceOf<T> = BondedPools::<T>::iter()
					.map(|(id, inner)| {
						T::Staking::total_stake(&BondedPool { id, inner }.bonded_account())
							.unwrap_or_default()
					})
					.reduce(|acc, balance| acc + balance)
					.unwrap_or_default();

				TotalValueLocked::<T>::set(tvl);`